src/ios/CDVLocation.m

```diff
@@ -169,199 +180,221 @@ - (void)_stopLocation
     }
 }
 
-- (void)locationManager:(CLLocationManager*)manager
-    didUpdateToLocation:(CLLocation*)newLocation
-           fromLocation:(CLLocation*)oldLocation
+- (void)locationManager:(CLLocationManager *)manager
+    didUpdateToLocation:(CLLocation *)newLocation
+           fromLocation:(CLLocation *)oldLocation
 {
-    CDVLocationData* cData = self.locationData;
-
+    CDVLocationData *cData = self.locationData;
     cData.locationInfo = newLocation;
+
+    [self processLocationCallbacks];
+
+    if (self.locationData.watchCallbacks.count > 0) {
+        [self processWatchCallbacks];
+    } else {
+        [self _stopLocation];
+    }
+}
+
+- (void)processLocationCallbacks
+{
     if (self.locationData.locationCallbacks.count > 0) {
-        for (NSString* callbackId in self.locationData.locationCallbacks) {
+        for (NSString *callbackId in self.locationData.locationCallbacks) {
             [self returnLocationInfo:callbackId andKeepCallback:NO];
         }
 
         [self.locationData.locationCallbacks removeAllObjects];
     }
-    if (self.locationData.watchCallbacks.count > 0) {
-        for (NSString* timerId in self.locationData.watchCallbacks) {
-            [self returnLocationInfo:[self.locationData.watchCallbacks objectForKey:timerId] andKeepCallback:YES];
-        }
-    } else {
-        // No callbacks waiting on us anymore, turn off listening.
-        [self _stopLocation];
+}
+
+- (void)processWatchCallbacks
+{
+    for (NSString *timerId in self.locationData.watchCallbacks) {
+        NSString *callbackId = [self.locationData.watchCallbacks objectForKey:timerId];
+        [self returnLocationInfo:callbackId andKeepCallback:YES];
     }
 }
 
-- (void)getLocation:(CDVInvokedUrlCommand*)command
+- (void)getLocation:(CDVInvokedUrlCommand *)command
 {
     [self.commandDelegate runInBackground:^{
         __weak __typeof(self) weakSelf = self;
-        NSString* callbackId = command.callbackId;
+        NSString *callbackId = command.callbackId;
         BOOL enableHighAccuracy = [[command argumentAtIndex:0] boolValue];
 
         [self isLocationServicesEnabledWithCompletion:^(BOOL enabled) {
             __strong __typeof(weakSelf) strongSelf = weakSelf;
             if (!strongSelf) {
                 return;
             }
```

Comment:
Before, this was changed to `if (!strongSelf) return`. Maybe we could keep things similar here?